core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,140 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
     test_can_play_entity.__wrapped__ = True
 
     return test_can_play_entity
+
+
+def can_view_submitted_voiceover_applications(handler):
+    """Decorator to check whether the user can view submitted voiceover
+    applications.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to view submitted voiceover applications.
+    """
+
+    def test_can_view_submitted_voiceover_applications(self, **kwargs):
+        """Checks if the user can view submitted voiceover applications.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        return handler(self, **kwargs)
+
+    test_can_view_submitted_voiceover_applications.__wrapped__ = True
+
+    return test_can_view_submitted_voiceover_applications
+
+
+def can_review_voiceover_application(handler):
+    """Decorator to check whether the user can review voiceover applications.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to review voiceover application.
+    """
+
+    def test_can_review_voiceover_application(
+            self, **kwargs):
+        """Checks if the user can review voiceover application.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+            UnauthorizedUserException: The user does not have
+                credentials to voiceover an exploration.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        user_actions_info = user_services.UserActionsInfo(self.user_id)
+        if (
+                role_services.ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION not in
+                user_actions_info.actions):
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to review voiceover '
+                'applications.')
+
+        return handler(self, **kwargs)
+
+    test_can_review_voiceover_application.__wrapped__ = True
+
+    return test_can_review_voiceover_application
+
+
+def can_submit_voiceover_application(handler):
```

Comment:
Maybe make this "can_submit_new_voiceover_application" to make it clearer that this is only handling a new submission and not an already-existing one.

Reply:
Done!